### ISSUE TITLE
Resample-mode crops return NaN when an output axis is 1 pixel

### ISSUE THREAD

**Issue by RizgarOzan** [kornia/kornia#4751]
### What happened?

Cropping to a size with a 1-pixel axis in resample mode returns an all-NaN output, with no error. `CenterCrop`, `RandomCrop` (`cropping_mode="resample"`) and the functional `center_crop` all do it. `CenterCrop(..., cropping_mode="slice")` is fine.

The four corners of a 1-pixel-high (or wide) box are collinear, so `get_perspective_transform(src, dst)` has no unique solution and returns a NaN matrix, which then goes into the warp. The 3D crops hit the same thing, but there the solver raises `_LinAlgError` instead (#4705, PR #4732). A crop only needs a translation (`dst = src - top_left`), so the matrix could be built directly like in #4732.

### Minimal reproduction

```python
import torch
import kornia.augmentation as K
from kornia.geometry.transform import center_crop

x = torch.rand(2, 1, 4, 5)
for size in [(1, 3), (3, 1), (1, 1)]:
    print(
        size,
        torch.isnan(K.CenterCrop(size, cropping_mode="resample")(x)).any().item(),
        torch.isnan(K.RandomCrop(size, cropping_mode="resample")(x)).any().item(),
        torch.isnan(center_crop(x, size)).any().item(),
    )
# (1, 3) True True True
# (3, 1) True True True
# (1, 1) True True True
```

`K.CenterCrop((1, 3), cropping_mode="resample")` also stores an all-NaN `transform_matrix`.

### Environment

kornia main (8b03a808), torch 2.14.0+cpu, Python 3.14, Windows 11.

### Anything else?

I can open the PR. I'd wait for #4732 first so both crops use the same approach, unless you'd rather I go ahead. Found while working on #4705 with an AI assistant; I ran the repro above myself.
